torch/lib/THS/generic/THSTensor.c

```diff
@@ -261,18 +326,27 @@ THTensor *THSTensor_(toDense)(THSTensor *self) {
   indices = THLongTensor_data(indices_);
   indskip = THLongTensor_size(indices_, 1); // To index indices
 
-  #pragma omp parallel for private(k, index)
+  #pragma omp parallel for private(k, d, l, index)
   for (k = 0; k < nnz; k++) {
-    for (d = 0, index = 0; d < ndim; d++)
+    for (d = 0, index = 0; d < nDimI; d++)
       index = sizes[d] * index + indices[d * indskip + k];
-    other[index] = values[k];
+    for (l = 0; l < blocksize; l++) {
+      other[index * blocksize + l] = values[k * blocksize + l];
+    }
   }
 
   THTensor_(free)(values_);
   THLongStorage_free(storage);
   return other_;
 }
 
+void THSTensor_(copy)(THSTensor *self, THSTensor *src) {
```

Comment:
Is this also an internal method? I think the usual semantics of copy is that if the tensors weren't sharing the data before, they won't be sharing it afterwards too.